### ISSUE TITLE
hasfacl outcome incorrect due to http-user grep?

### ISSUE THREAD

**Issue by pascal08** [deployphp/deployer#1557]
| Q                 | A
| ----------------- | ---
| Issue Type        | Bug/Question
| Deployer Version  | 6.0.5
| Local Machine OS  | Ubuntu 16.04
| Remote Machine OS | Ubuntu 16.04

### Description 
I think something goes wrong when determining if a ACL is set for a certain folder and here is why:

Looking at [this line](https://github.com/deployphp/deployer/blob/master/recipe/deploy/writable.php#L78) I see the ACL check is dependent on the $httpUser variable. In my case $httpUser is detemined by [this line](https://github.com/deployphp/deployer/blob/master/recipe/deploy/writable.php#L27). When executing this command direct on my remote server it returns: ```www-data nginx```. Plugging this variable in the getfacl grep command would return nothing which I think is incorrect. 

### Steps to reproduce
N/A

### Content of `deploy.php`
N/A

### Output log
N/A

**Comment by pascal08** [comment]
I didn't notice there were two space at [this line](https://github.com/deployphp/deployer/blob/master/recipe/deploy/writable.php#L27)

Maybe better for readability to use this instead?

```
ps axo user,comm | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | head -1 | cut -d ' ' -f1
```

**Comment by antonmedv** [comment]
Yes, agree!

**Comment by pascal08** [comment]
@antonmedv Great. I will make a pull request in a minute.

I'm still wondering...

This is the output from Deployer:
```
The command "cd /*** && (setfacl -RL -m u:"www-data":rwX -m u:`whoami`:rwX var)" failed.
```

Running [this command](https://github.com/deployphp/deployer/blob/master/recipe/deploy/writable.php#L78) by hand on my remote server returns `1` which should skip line 81 and 82 right?

**Comment by antonmedv** [comment]
Yes, but I think you can try sudo for setfacl.

**Comment by pascal08** [comment]
You mean setting 'writable_use_sudo' to true?

**Comment by bkosborne** [comment]
I have this problem as well. For some reason the command that checks if setfacl was already run on the directory is not returning "1" like it should. I think it's returning nothing. But when I run the command manually myself it returns a 1 as expected. I'm not quite sure what the problem is though!

**Comment by bkosborne** [comment]
Note that the directory that produced a failure for me was a directory that's symlinked into the one I'm actually trying to make writable. Maybe that's the problem.

**Comment by joelpittet** [comment]
I ran into this problem too.  Something to do with AWS EFS mounts I bet.
chown + sudo worked.
```
set('writable_mode', 'chown');
set('writable_use_sudo', TRUE);
```

**Comment by hisem** [comment]
Same problem here, and I also do not wish to give sudo rights to my deployment user. @pascal08 did you find another solution?

**Comment by hisem** [comment]
@pascal08 Ok, I found out what the problem was: 

I was getting the `setfacl` error `Operation not permitted` on `var/log`, but the `getfacl` command was running on `var`, which is automatically added to the list of `writable_dirs`:

```
//recipe/symfony4.php l.12
set('shared_dirs', ['var/log', 'var/sessions']);
set('shared_files', ['.env']);
set('writable_dirs', ['var']);
```

What is happening is that the `getfacl` returns "0" on `var` because it is not in the shared_dirs (so it is newly created in the deploy). It passes the `!$hasfacl` condition, then runs `setfacl` on it and its children (as it runs recursively using -R) which contains shared_dirs (`var/log` and `var/sessions`, by default), which in turn contain files that have been created by the http user. 

_`setfacl` can't be run on a folder you are not the owner of, even if you have rwx rights, leading to the `Operation not permitted` error._

My solution was to reset the writable_dirs and shared_dirs as follow:

```
set('shared_dirs', ['var/log', 'var/sessions']);
set('writable_dirs', ['var/log', 'var/sessions', 'var/cache']); //you can add other children of 'var' here if you have any other
```

**removing the 'var' directory from `writable_dirs` and replacing it with its children.**

@antonmedv In light of this, should the default `writable_dirs` be changed?

**Comment by antonmedv** [comment]
`var` stands for variable. Maybe do this instead of: 
```php
set('shared_dirs', ['var']);
set('writable_dirs', ['var']);
```

**Comment by hisem** [comment]
@antonmedv I thought of doing that, but the var directory contains the cache directory. 
Should the cache be "shared"? 🤔 Would it have any side effects? Maybe for rollbacks?

**Comment by antonmedv** [comment]
Hmmm, I see. Need to investigate about using cache between deploys. Maybe it's a good thing to do.

**Comment by antonmedv** [comment]
Should be fixed in #1904
